tests/conftest.py

```diff
@@ -1300,57 +1300,69 @@ def get_inband(self, ctnname):
-            vethintf = int(ncv[ctndir].split("eth")[1])
-            cfgdir = cwd + "/virtual_chassis/" + ctndir
-            cfgfile = cfgdir + "/default_config.json"
-            ctnname = ""
-            with open(cfgfile, "r") as cfg:
-                defcfg = json.load(cfg)["DEVICE_METADATA"]["localhost"]
-                ctnname = defcfg["hostname"] + "." + self.ns
-            cfgfile = cwd + "/virtual_chassis/" + nbrctndir + "/default_config.json"
-            with open(cfgfile, "r") as cfg:
-                intfCfg = json.load(cfg)["INTERFACE"]
-                for key in intfCfg:
-                    nbraddr = ""
-                    if key.lower().startswith(nbrintf):
-                        intfaddr = re.split("/|\\|", key)
-                        if len(intfaddr) > 1:
-                            nbraddr = intfaddr[1]
-                        if nbraddr == "":
+            return instance_to_neighbor_map
+
+        working_dir = os.getcwd()
+        for conn, endpoints in self.virt_topo["neighbor_connections"].items():
+            chassis_instance = conn.split('-')[0]
+            neighbor_instance = conn.split('-')[1]
+
+            chassis_config_dir = os.path.join(working_dir, "virtual_chassis", chassis_instance)
+            chassis_config_file = os.path.join(chassis_config_dir, "default_config.json")
+            instance_container_name = ""
+            with open(chassis_config_file, "r") as cfg:
+                device_info = json.load(cfg)["DEVICE_METADATA"]["localhost"]
+                instance_container_name = device_info["hostname"] + "." + self.ns
+
+            chassis_veth_intf = int(endpoints[chassis_instance].split("eth")[1])
+            chassis_host_intf = f"Ethernet{(chassis_veth_intf - 1) * 4}"
+
+            chassis_config_file = os.path.join(working_dir, "virtual_chassis", neighbor_instance, "default_config.json")
+            with open(chassis_config_file, "r") as cfg:
+                intf_config = json.load(cfg)["INTERFACE"]
+                for key in intf_config:
+                    neighbor_address = ""
+                    if key.lower().startswith(f"{chassis_host_intf}|".lower()):
+                        host_intf_address = re.split("/|\\|", key)
+
+                        if len(host_intf_address) > 1:
+                            neighbor_address = host_intf_address[1]
+
+                        if neighbor_address == "":
                             continue
-                        if ctnname not in neighs_by_ctn:
-                            neighs_by_ctn[ctnname] = []
-                        neighs_by_ctn[ctnname].append((vethintf - 1, nbraddr))
-        return neighs_by_ctn
+
+                        if instance_container_name not in instance_to_neighbor_map:
+                            instance_to_neighbor_map[instance_container_name] = []
+
+                        instance_to_neighbor_map[instance_container_name].append((chassis_veth_intf - 1,
```

Comment:
instance_to_neighbor_map[instance_container_name].append((neighbor_veth_intf - 1, neighbor_address)

Reply:
Makes sense, thank you!

Reply:
may be a comment can help on why we return some info about the neighbor-instance and some about the chassis-instance. 
Neighbor info is there so that verify command can run ping to the neighbor to check connectivity.

Reply:
That makes sense! Rather than doing that, I separated the chassis port bring-up part from the neighbor info methods. 

`get_topo_neigh` and `handle_neighconn` should be the same as before now minus the variable names, all the chassis port admin status handling is in `get_chassis_instance_port_statuses` and `handle_chassis_connections` (it makes more sense to get that info from the chassis instance config files anyways :)).